agreement/msgp_gen.go

```diff
@@ -6563,7 +6869,7 @@ func (z *rootRouter) UnmarshalMsg(bts []byte) (o []byte, err error) {
 				(*z).Children = make(map[round]*roundRouter, zb0009)
 			}
 			for zb0009 > 0 {
-				var zb0001 round
+				var zb0001 basics.Round
```

Comment:
That's actually a great question. We should verify this

Reply:
With rootRouteBasic defined as 
```
type rootRouterBasic struct {
	_struct struct{} `codec:","`

	root         actor    // playerMachine   (not restored: explicitly set on construction)
	proposalRoot listener // proposalMachine
	voteRoot     listener // voteMachine

	ProposalManager proposalManager
	VoteAggregator  voteAggregator

	Children map[basics.Round]*roundRouter `codec:"Children,allocbound=-"`
}
```

i.e. exactly the same but using basics.Round as the key instead of the alias the following test passes. 

```
func TestMsgpTypeAliasCompat(t *testing.T) {
	partitiontest.PartitionTest(t)

	randomRound := rand.Uint64()
	a := rootRouter{Children: map[round]*roundRouter{round(randomRound): {}}}

	b := rootRouterBasic{Children: map[basics.Round]*roundRouter{basics.Round(randomRound): {}}}

	require.NotEqual(t, a, b)

	aEnc := protocol.Encode(&a)
	bEnc := protocol.Encode(&b)

	require.Equal(t, aEnc, bEnc)

	var aDecA, bDecA rootRouter
	var aDecB, bDecB rootRouterBasic

	err := protocol.Decode(aEnc, &aDecA)
	require.NoError(t, err)
	err = protocol.Decode(bEnc, &bDecA)
	require.NoError(t, err)
	require.Equal(t, aDecA, bDecA)

	err = protocol.Decode(aEnc, &aDecB)
	require.NoError(t, err)
	err = protocol.Decode(bEnc, &bDecB)
	require.NoError(t, err)
	require.Equal(t, aDecB, bDecB)
```

Reply:
looks good, thanks